### ISSUE TITLE
Poor Lessson

### ISSUE THREAD

**Issue by davidjkim810** [learn-co-curriculum/react-router-nested-routes-code-along#8]
This lesson is very confusing and hard to follow. So after all this 

"Master Detail Without Routes
Have you ever used Apple's Messages app for your Mac? How about Gmail? What about YouTube? All of those apps use some version of a "Master-Detail" interface. This is when there is something pertaining to the entire resource, such as a list of all messages, videos, or emails, and some more detailed display of a specific item or action on another portion of the screen. Clicking on a new item in the list changes which item we have selected.

Nesting
With React-Router, we can make the master-detail pattern by making our components children of each other. Take YouTube for example. Let's pretend that visiting /videos displays a list of videos. Clicking on any video keeps our list of videos on the page, but also displays details on the selected video. This should be updated by the URL - the URL should have changed to /videos/:videoId. The VideoDetail in this case is a 'Nested Component' of '/videos' - it will always have the list rendered before it."

So what exactly is a master detail pattern? 

Also, ALL of a sudden we go from rendering one line components. To this:

"<Route path='/movies' render={routerProps => <MoviesPage {...routerProps} movies={this.state.movies}/>} />"

Also

"The easiest way to handle this is to use the render. The render attribute of Route takes an anonymous function, and since Route is passing its own props automatically, they are available as the argument for the function, called routerProps here. Since the function is simply returning a component, we can then pass in both routerProps and this.state.movies as props to the MoviesPage component."

What would be more helpful would be a video walk through of the lab. With clear explanations on each part.

**Comment by atxrenegade** [comment]
Super, unclear and difficult to follow from beginning to finish.  Requires constant jumping back and forth between code samples and content in an attempt to make sense of a very dense explanation.

**Comment by hankeliu2015** [comment]
should explain why use spread operator for routerProps?

**Comment by maxwellbenton** [comment]
Hi @davidjkim810, @atxrenegade, and @hankeliu2015,

Thank you for providing this feedback. We've updated this lesson's instructions to be more detailed and explicit in each of the steps required to created nested routes. The lesson now goes into further detail on the concept of a "master-detail" interface and loops back around at the end to reinforce the concept; it includes explanations into why we use the `render` instead of `component` prop in these situations, what `routerProps` is along with a brief explanation of the use of the spread operator to pass props. We've also added additional code snippets to make the steps clearer and a final summary of what is required to set up nested routing. These changes should help to reduce the confusion caused by this lesson.

We apologize for any frustration this lesson caused you, but we appreciate you taking the time to provide this feedback and help improve our curriculum for future students!

Thank you!
